packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -140,16 +140,15 @@ exports.sourceNodes = async (
   downloadingFilesActivity.start()
 
   // Download all files (await for each pool to complete to fix concurrency issues)
-  await asyncPool(
-    concurrentFileRequests,
-    [...nodes.values()].filter(isFileNode),
-    async node => {
+  const fileNodes = [...nodes.values()].filter(isFileNode)
+  if (fileNodes.length) {
+    await asyncPool(concurrentFileRequests, fileNodes, async node => {
       await downloadFile(
         { node, store, cache, createNode, createNodeId },
         pluginOptions
       )
-    }
-  )
+    })
+  }
 
   downloadingFilesActivity.end()
```

Comment:
Please move this inside `if (fileNodes.length)` conditional code path

Reply:
@pieh Thank you for catching that. I've made those changes.